File: Snooper/src/main/java/com/prateekj/snooper/customviews/AdapterDataAppendObserver.java
package com.prateekj.snooper.customviews;

import androidx.recyclerview.widget.RecyclerView;

public class AdapterDataAppendObserver extends RecyclerView.AdapterDataObserver {

  private PageAddedListener listener;

  public AdapterDataAppendObserver(PageAddedListener listener) {
    this.listener = listener;
  }

  @Override
  public void onChanged() {
    super.onChanged();
    listener.onPageAdded();
  }
}

File: Snooper/src/main/java/com/prateekj/snooper/customviews/AdapterDataAppendObserver.kt
package com.prateekj.snooper.customviews

import androidx.recyclerview.widget.RecyclerView

class AdapterDataAppendObserver(private val listener: PageAddedListener) :
  RecyclerView.AdapterDataObserver() {

  override fun onChanged() {
    super.onChanged()
    listener.onPageAdded()
  }
}
